include/mg_procedure.h

```diff
@@ -1405,39 +1405,74 @@ int mgp_must_abort(struct mgp_graph *graph);
 
 /// @}
 
-/// @name Kafka message API
-/// Currently the API below is for kafka only but in the future
-/// mgp_message and mgp_messages might be generic to support
-/// other streaming systems.
+/// @name Stream Source message API
+/// API for accessing specific data contained in a mgp_message
+/// used for defining transformation procedures.
+/// Not all methods are available for all stream sources
+/// so make sure that your transformation procedure can be used
+/// for a specific source, i.e. only valid methods are used.
 ///@{
 
-/// A single Kafka message
+/// A single Stream source message
 struct mgp_message;
 
-/// A list of Kafka messages
+/// A list of Stream source messages
 struct mgp_messages;
 
+/// Stream source type.
+enum mgp_source_type {
+  KAFKA,
+  PULSAR,
+};
+
+/// Get the type of the stream source that produced the message.
+enum mgp_error mgp_message_source_type(struct mgp_message *message, enum mgp_source_type *result);
+
 /// Payload is not null terminated and not a string but rather a byte array.
 /// You need to call mgp_message_payload_size() first, to read the size of
 /// the payload.
+/// Supported stream sources:
+///   - Kafka
+///   - Pulsar
```

Comment:
Because this is connected to types you can define `CREATE KAFKA ...` and `CREATE PULSAR...`.
In our case, we do support Redpanda but using the Kafka stream source type.
But nice catch, we need to remind ourselves to tell people that they can use it.